Updated PySide to PySide2.

tsalemink opened the pull request.

tsalemink committed:
Updated PySide to PySide2.

tsalemink committed:
Updated old import for custom widget.

tsalemink committed:
Updated handling of plugin dependencies to work with updated MAP-Client version.

tsalemink committed:
Removed files that are no longer being used.

tsalemink committed:
Python 2to3.py changes.

tsalemink committed:
Updated datamapstep.

Updated datamapstep to allow the user to specify the model coordinates field and data coordinates field, rather than having to hard code these identifiers.

tsalemink committed:
Updated coordinate field combo-boxes.

The QComboBoxes now produce a warning pop-up if the user chooses an invalid coordinate field. The QComboBoxes also have dummy elements; note that the current implementation is not ideal since the method combo_box.setPlaceholderText() currently appears to be broken.

tsalemink committed:
Updated step to correctly display the user's current field choices.

hsorby commented on mapclientplugins/datamapstep/model/mapper.py:
```diff
@@ -55,6 +55,59 @@ def get_data_projection_error_field(self):
+        if field_name == "---":
+            self._data_coordinates_field_name = None
+            self._data_coordinates_field = None
+            return
```
I think you might need to change the name of fields here.

hsorby commented on mapclientplugins/datamapstep/model/mapper.py:
```diff
@@ -55,6 +55,59 @@ def get_data_projection_error_field(self):
+
+        field = self._field_module.findFieldByName(self._model_coordinates_field_name)
+        finite_element_field = field.castFiniteElement()
+        if not (finite_element_field.isValid() and (finite_element_field.getNumberOfComponents() == 3)):
+            self._data_coordinates_field_name = None
+            self._data_coordinates_field = None
```
Same applies here.

hsorby commented on mapclientplugins/datamapstep/model/mappingmodel.py:
```diff
@@ -79,29 +79,25 @@ def _autorange_spectrum(self):
-            data_points.setCoordinateField(data_coordinate)
             point_attr = data_points.getGraphicspointattributes()
-            point_attr.setBaseSize([10, 10, 10])
+            point_attr.setBaseSize([200, 200, 200])
```
This may need to become a configurable item in the user interface, it is related to the size of the mesh.  If someone has a mesh that is very small then these points are going to be very big in comparison.

hsorby commented on mapclientplugins/datamapstep/model/mapper.py:
```diff
@@ -55,6 +55,59 @@ def get_data_projection_error_field(self):
+            raise ValueError("The selected data coordinates field is not valid.")
+        self._data_coordinates_field = field
+
+    def get_field_list(self):
```
I think this method should only get finite element fields of dimension 3, this would then allow dropping this check in the above methods.

tsalemink committed:
Minor fixes.

Updated the step to allow the user to specify the size of the glyphs used to visualize the data points. Updated the way the list of possible coordinate fields are generated.
